pkg/portal/cluster/machines.go

```diff
@@ -59,6 +68,55 @@ func (c *client) Machines(ctx context.Context) (*MachineListInformation, error)
 	return c.fetcher.Machines(ctx)
 }
 
+// VM Allocation Status Stuff
+type VMAllocationStatus map[string]string
+
+func (c *client) VMAllocationStatus(ctx context.Context) (VMAllocationStatus, error) {
+	return c.fetcher.VMAllocationStatus(ctx)
+}
+
+func (f *realFetcher) VMAllocationStatus(ctx context.Context) (VMAllocationStatus, error) {
+	env := f.azureSideFetcher.env
+	subscriptionDoc := f.azureSideFetcher.subscriptionDoc
+	clusterRGName := f.azureSideFetcher.resourceGroupName
+	fpAuth, err := env.FPAuthorizer(subscriptionDoc.Subscription.Properties.TenantID, env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// Getting Virtual Machine resources through the Cluster's Resource Group
+	computeResources, err := newResourceClientFunction(env.Environment(), subscriptionDoc.ID, fpAuth).ListByResourceGroup(ctx, clusterRGName, "resourceType eq 'Microsoft.Compute/virtualMachines'", "", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	vmAllocationStatus := make(VMAllocationStatus)
+	virtualMachineClient := newVirtualMachineClientFunction(env.Environment(), subscriptionDoc.ID, fpAuth)
+	for _, res := range computeResources {
+		var vmName, allocationStatus string
+		vm, err := virtualMachineClient.Get(ctx, clusterRGName, *res.Name, mgmtcompute.InstanceView)
+		if err != nil {
+			f.log.Warn(err) // can happen when the ARM cache is lagging
+			// armResources = append(armResources, arm.Resource{
+			// 	Resource: res,
+			// })
+			continue
+		}
+
+		vmName = *vm.Name
+		instanceViewStatuses := vm.InstanceView.Statuses
+		for _, status := range *instanceViewStatuses {
+			if strings.Contains(*status.Code, "PowerState") {
+				allocationStatus = *status.Code
+			}
+		}
+
+		vmAllocationStatus[vmName] = allocationStatus
```

Comment:
PowerState is at least some state as it comes in the for of "PowerState/{somestate}" for example "PowerState/running" and I take only the status code which has "PowerState" in it. 
Since it is looping through the VMs .. there is no chance of a VMs powerstate getting overridden.